api/filter/accesslog.proto

```diff
@@ -2,31 +2,59 @@ syntax = "proto3";
 
 package envoy.api.v2.filter;
 
+import "api/address.proto";
 import "api/base.proto";
 
 import "google/protobuf/duration.proto";
 import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/wrappers.proto";
 
-message AccessLogEntry {
-  // The HTTP request method (RFC 7231/2616)
-  RequestMethod request_method = 1;
+// Envoy access logs describe incoming interaction with Envoy over a fixed
+// period of time, and typically cover a single request/response exchange,
+// (e.g. HTTP), stream (e.g. over HTTP/gRPC), or proxied connection (e.g. TCP).
+// Except where explicitly declared otherwise, all fields describe
+// *downstream* interaction between Envoy and a connected client.
+// Fields describing *upstream* interaction will explicitly include `upstream`
+// in their name.
+message AccessLogCommon {
+  // This field indicates the rate at which this log entry was sampled.
+  // Valid range is (0.0, 1.0].
+  double sample_rate = 1;
```

Comment:
Do you want to give adding constraints annotations here a shot? See https://github.com/envoyproxy/data-plane-api/blob/master/api/filter/fault.proto#L25 for example, https://github.com/lyft/protoc-gen-validate/blob/master/validate/validate.proto for the constraints defs.